pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -2565,10 +2612,73 @@ func (c *Cloud) getTaggedSecurityGroups() (map[string]*ec2.SecurityGroup, error)
 	return m, nil
 }
 
+// Almost identical to updateInstanceSecurityGroupsForLoadBalancer, but it only adds ssg rules to
+// instances' inbound rules.
+func (s *AWSCloud) updateInstanceSharedSecurityGroups(ssgID string, allInstances []*ec2.Instance) error {
+	if s.cfg.Global.DisableSecurityGroupIngress || s.cfg.Global.EnableSharedSecurityGroupIngress == false {
```

Comment:
@justinsb you are right. that was for double safeguarding basically. They both can't be true, because it will fast fail if someone sets both of the flags to true. But I will delete it.